Analysis/Core/src/TrackSelection.cxx

```diff
@@ -0,0 +1,63 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+//
+// Class for track selection
+//
+
+#include "Analysis/TrackSelection.h"
+#include <TH1F.h>
+
+using namespace o2;
+using namespace o2::framework;
+
+TrackSelection::TrackSelection()
+    : mMinPt{0.}, mMaxPt{1e10}, mMinEta{0.}, mMaxEta{1e10}, mMinNClustersTPC{0},
+      mMinNCrossedRowsTPC{0}, mMinNClustersITS{0}, mMaxChi2PerClusterTPC{1e10},
+      mMaxChi2PerClusterITS{1e10},
+      mMinNCrossedRowsOverFindableClustersTPC{1e10}, mMaxDcaXY{1e10},
+      mMaxDcaZ{1e10}, mRequireITSRefit{false}, mRequireTPCRefit{false} {}
+
+bool TrackSelection::IsSelected(
+    soa::Join<aod::Tracks, aod::TracksCov, aod::TracksExtra>::iterator const
+        &track) {
+  if (track.pt() >= mMinPt && track.pt() < mMaxPt && track.eta() >= mMinEta &&
+      track.eta() < mMaxEta && track.tpcNClsFound() >= mMinNClustersTPC &&
+      track.tpcNClsCrossedRows() >= mMinNCrossedRowsTPC &&
+      track.tpcCrossedRowsOverFindableCls() >=
+          mMinNCrossedRowsOverFindableClustersTPC &&
+      (track.itsNCls() >= mMinNClustersITS) &&
+      (track.itsChi2NCl() < mMaxChi2PerClusterITS) &&
+      (track.tpcChi2Ncl() < mMaxChi2PerClusterTPC) &&
+      (mRequireITSRefit && (track.flags() & 0x4)) &&
+      (mRequireTPCRefit && (track.flags() & 0x40)) &&
+      FulfillsITSHitRequirements(track.itsClusterMap())) {
+    return true;
```

Comment:
Even if long, I'd split this into several statements, otherwise it will become very unreadable in the future. This will also allow us to add some histograms of cut correlations.

Reply:
Sure, I can split it. I put it like this because I assumed that this will be replaced anyway by some logic involving the Framework filters.

Reply:
Yes, you are right. So ignore my comment for now.